### ISSUE TITLE
User flow for implementing model-based detection system into QF rounds

### ISSUE THREAD

**Issue by laurenluz** [Giveth/giveth-dapps-v2#4247]
We are adding a 2-part user verification model for qf eligibility. Backend is explained in https://github.com/Giveth/giveth-dapps-v2/issues/4221

This issue is for design. The user flow is essentially:
- At the start of a new QF round, every donor must verify/re-verify their eligibility
- The first "check" is called either by a button click "check your eligibility" or by the user making their first donation. In the backend, we check if their address passes... if they do, the user sees "you're eligible!" and no other info about passport/stamps, etc.
- If the user fails the 1st check (with the model based system), they are passed through our gitcoin passport flow, where they can verify their stamps

We will have to consider:
- banners on QF page
- donation success page eligibility box
- user profile
- giveth.io/passport page (we may want to revamp this into a qf eligibility page)

@mosaeedi @koday1 cc @CarlosQ96

**Comment by mosaeedi** [comment]
## Here is the first design draft

Please take a look at the proposed structure and designs for the Profile page.


# Structure 

So with the new method in mind, I create two sections for the profile page.

**1. User Verification**
On the first section, User can verify with the MBD method, or [in case they couldn't pass the MBD] use Gitcoin Passport.


**2. QF Donation Eligibility**
This section is more like an informative section just to make it crystal clear for user either they are eligible or not.


# Designs 
Now let's dive into the designs!


## User A
Users A, are those who don't have a connected Gitcoin Passport and are Not verified using the MBD method.
For these users, they can only see the MBD card and the Gitcoin passport card is grayed out and disabled.

<img width="811" alt="Screenshot 2024-06-11 at 4 55 04 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/e327d5ad-7c5f-4797-9f40-c773078a8ee5">

### User A-1 [MBD Verified]

When they pass the MBD check successfully, the Gitcoin Passport is get enabled, but on the QF eligibility card, they can see that, they are already eligible for QF matching.

### User A-2 [MBD Failed]

In case the user fails the MBD check, the Gitcoin passport is get enabled as an alternative way for user to verify.

<img width="815" alt="Screenshot 2024-06-11 at 5 02 10 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/abd8b2aa-f087-4d27-843d-ece274e4ca67">

### User A-3 [MBD Failed, Passport Checked]
And when they passed the Gitcoin check...

<img width="889" alt="Screenshot 2024-06-11 at 5 05 15 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/c47af3ee-b113-41e5-8cc8-2cbb6ef719ae">



### User B
User B, are those who already connected their Gitcoin Passport 

<img width="870" alt="Screenshot 2024-06-11 at 5 06 22 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/84499738-5586-4771-8bfe-5064c2ed7e14">




### User C
User C, are those who passed the MBD check and also connected their Passport before.

<img width="780" alt="Screenshot 2024-06-11 at 5 07 34 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/1c0210e7-f970-42b9-a5b1-1e641c3166ee">

**Comment by koday1** [comment]
Thank @mosaeedi that's looking great! I just shared your comment w/ @CarlosQ96 as he's working on the backend API integration. 

One thought - I wonder if users will be confused if we name it "Quick identity check", since it's not really related to identity. Maybe we can name it "Quick on-chain activity check" or something similar? What do you think @laurenluz ?

**Comment by laurenluz** [comment]
Hey @mosaeedi! This looks pretty good but tbh I think we can make it a lot more simplified for the user. Seeing all these parts "user verification" and having that separated from "qf eligibility" imo is not necessary because the user verification IS the eligibility, it's just the method by which they verify that changes. The advantage of the MBD system is that we can make thing a lot less complex for many users... so I would the design to be a lot simpler as well.

If you think it's ok, I would suggest removing the bottom box entirely and having the user see just one box basically... that changes in copy and appearance based on the following states. They would only see the gitcoin passport info (replacing the 1st box) if they do not pass the 1st check on the MBD. So the states would be:
(1) wallet not connect
(2) wallet connected but not yet checked w/ MBD (user sees button saying "Check eligibility")
(3) wallet connect MBD passed (button grayed out, user see label "QF eligible")
(4) wallet connected, MBD failed, gitcoin passport not yet checked (Copy is "We need a bit more information to check your uniqueness. Please verify your Gitcoin passport")
(5) wallet connected, MBD failed, gitcion passport checked and not passing 
(6) wallet connected, MBD failed, gitcoin passport check and passing (button grayed out, user sees lable "QF eligible")

For initial copy, I'd suggest:

Title for the page: QF Donor Eligibility
Subtext: Verify your identity to ensure your donations can be matching in quadratic funding rounds.

Box copy, e.g. for case (2) wallet connected by not yet checked w/ MBD
Box title: Quick Identity Check
Subtext: Verify your donor uniqueness with a quick check of your on-chain activity.
Button: Check eligibility

After the user passes, the label says "QF eligible".

What do you think? Also could you link to the figma?
cc @koday1

**Comment by laurenluz** [comment]
Aside from that, can you also help to outline the banner copy & the donation success page copies for each of the different states? I think we will need to make adjustments to the logic and it would help to make this all clear for our devs.

Also, thinking about this... should we keep all this verification information in the user profile? or do you think it would be better to make this a kind of replacement for the https://giveth.io/passport ?

And then we direct users to that page through out existing QF flow?

**Comment by mosaeedi** [comment]
Hey,
I've updated the designs and made some adjustments,

I've added the QF eligibility badge to the header instead of showing it at the bottom!
please take a look.

> (1) wallet not connect
[ No Change ]


> (2) wallet connected but not yet checked w/ MBD (user sees button saying "Check Eligibility")

<img width="1014" alt="Screenshot 2024-06-15 at 4 02 37 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/abb7f4fa-5c16-4177-961c-467617381c0a">

> (3) Wallet connect MBD passed (button grayed out, user see label "QF eligible")

<img width="1018" alt="Screenshot 2024-06-15 at 4 03 01 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/ffe6d2b4-92a5-4444-b0bb-bcf33cdd1a89">


> (4) wallet connected, MBD failed, Gitcoin passport not yet checked (Copy is "We need a bit more information to check your uniqueness. Please verify your Gitcoin passport")

<img width="1040" alt="Screenshot 2024-06-15 at 4 04 15 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/f568cef0-0859-451b-b19d-e4a65762a508">


> (5) wallet connected, MBD failed, gitcion passport checked and not passing
[ I'm not sure what I have to do for this step ]


> (6) wallet connected, MBD failed, gitcoin passport check and passing (button grayed out, user sees lable "QF eligible")

<img width="814" alt="Screenshot 2024-06-15 at 4 05 46 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/5857b42b-9296-452c-b2a5-c2770b166c9a">

**Comment by mosaeedi** [comment]
Take a look here -> [Figma Link](https://www.figma.com/design/0YfkSvCVtc3Zhh7dBjoMuB/My-Account?node-id=94-1471&t=kDipFS7z5kcXLEPU-4)

**Comment by mosaeedi** [comment]
Here is the mini banner for the Projects list/view project page
I've updated some and disabled some(Grayed out).
Please review all of them and let me know if we need to add any more banners or won't need any of them.

take a look here -> [Figma Link](https://www.figma.com/design/Zjrx4AB4l5k9GzjjfLDIwR/Project-List?node-id=6004-6078&t=6p43AKqKwsx9U7PO-4)

**Comment by mosaeedi** [comment]
And the last part!

Success page

Take a look here -> [Figma Link](https://www.figma.com/design/qXxAEsN3jePyRqBcHR9Ihq/Donation?node-id=6001-5432&t=hDoSoYSLXcUHhIoO-4)

**Comment by CarlosQ96** [comment]
@mosaeedi @koday1 let me know if the design is done so @Meriem-BM can start on it.

**Comment by mosaeedi** [comment]
Here is the final designs for the **My account / Overview tab** 


**My account / Overview** 

[Figma link](https://www.figma.com/design/0YfkSvCVtc3Zhh7dBjoMuB/My-Account?node-id=178-5170)


The screen on the left shows the overview tab with the new QF card.
On the right side, you can see the QF card in all different states.


<img width="537" alt="Screenshot 2024-06-24 at 6 16 49 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/d54634d4-f6aa-4150-b213-6695d6e85dc3">

**Comment by mosaeedi** [comment]
And the **QF eligibility**  page, aka Passport page(before).


[Link to Figma](https://www.figma.com/design/0YfkSvCVtc3Zhh7dBjoMuB/My-Account?node-id=179-43768)

The QF Eligibility page on the left! and the different states of the QF Eligibility card on the right. 

Also you can look at the prototype here to better understand how it works.

[Video prototype on Youtube](https://www.youtube.com/watch?v=LMiyZGGEXYM)

<img width="438" alt="Screenshot 2024-06-24 at 6 18 42 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/6ea1feaa-d23d-4ff1-b679-f8f758593e7a">

**Comment by laurenluz** [comment]
@mosaeedi let's finalize the banners before we pass this to devs.

**Comment by laurenluz** [comment]
also @koday1 @CarlosQ96 @Meriem-BM we have the made sure we don't merge this issue until the GIV-earth round is over. we can introduce a big change of eligibility like this mid-round, and it's ofc not going to be finalized tested and merged before tomorrow morning.

**Comment by mosaeedi** [comment]
Here is the QF banners!


[Link to Figma](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail?node-id=7013-1662&t=MYwjOncksHT85qCv-4)

CC: @laurenluz @koday1

**Comment by koday1** [comment]
@mosaeedi The banners look good! @Meriem-BM please see the new banners above from Mo, and let me know if you have any questions.

**Comment by laurenluz** [comment]
these banners need a few edits, please hold off @Meriem-BM @koday1

**Comment by laurenluz** [comment]
@mosaeedi I added some missing cases to the banners and make some edits to the copy [here](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail?node-id=7024-221&t=YGm4uz5xg5ZtEoYO-4).

Can you help to clean them up?

Also, we need to have different donation success page info corresponding to the different cases as well. I'm talking about the copy here... We must have in figma the designs for the different cases... but I don't know where. @mosaeedi 

![image](https://github.com/Giveth/giveth-dapps-v2/assets/75490971/97180328-ab29-4982-831b-b325953a31ae)

**Comment by mosaeedi** [comment]
## The QF banners

### 1. loading state

<img width="615" alt="Screenshot 2024-07-04 at 7 26 47 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/d73b53c2-ff00-4d8e-80d7-060ec2d76986">


### 2. Wallet is not connected

<img width="732" alt="Screenshot 2024-07-04 at 7 27 46 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/3fe55f3e-1c4c-4a95-9b56-06168444e7a7">

### not checked for QF eligibility yet

<img width="757" alt="Screenshot 2024-07-04 at 7 29 37 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/2404ad1b-0940-4198-bbb5-7469a12ed421">


###  Processing

<img width="751" alt="Screenshot 2024-07-04 at 7 34 35 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/39c2e073-fd73-4186-8a32-79e63f637236">


## Two possible scenarios here 

### 1. MBD check passed
 
### QF eligible
<img width="528" alt="Screenshot 2024-07-04 at 7 43 01 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/72c8b4e9-5590-41b0-880e-01dece9bd9f1">


### 2.MBD failed, Passport is required

<img width="800" alt="Screenshot 2024-07-04 at 7 44 21 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/ccd59b0d-564b-440e-91b0-d6260a741eb5">


### Connecting passport

When user tap on the **Add more info**  button, we present the modal below,


<img width="898" alt="Screenshot 2024-07-04 at 7 45 27 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/919c5ff4-9744-4939-a923-0e168e91bb2e">

## There are two possible scenarios here as well

### 1. Passport score is higher than the requirement

<img width="568" alt="Screenshot 2024-07-04 at 7 53 13 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/3a85fad6-1f49-4e9b-902b-ee3cacb2bb51">

**The banner will change to the QF eligible state too**
<img width="336" alt="Screenshot 2024-07-04 at 8 00 20 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/7cf238f3-78b1-431e-bb4c-1b3b431f6cab">



### 2. Passport score is not high enough
<img width="544" alt="Screenshot 2024-07-04 at 7 58 35 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/209ae2a5-282b-48f1-bdc7-472db5c3371f">

**The banner will change to**
<img width="630" alt="Screenshot 2024-07-04 at 8 08 28 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/4f288578-0bbe-4b2c-a003-361759120027">




---- 


# Error states

### QF eligibility check error

<img width="703" alt="Screenshot 2024-07-04 at 8 11 15 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/45c1589f-2303-48ab-b4c4-7867c2f17503">


### QF round is finished

<img width="536" alt="Screenshot 2024-07-04 at 8 11 37 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/89f239e9-fce1-49b5-ada2-195d8832c673">

**Comment by mosaeedi** [comment]
You can also take a look at the Designs here,

[Link to Figma](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail?node-id=7035-4340&t=LoXKw8ChzvCwsw69-4)

**Comment by laurenluz** [comment]
This is so fantastic @mosaeedi !! Thank you for the detailed breakdown.

@Meriem-BM can you take this over? Ready for you now ❤️

Cc @koday1

**Comment by maryjaf** [comment]
> Here is the final designs for the **My account / Overview tab**
> 
> **My account / Overview**
> 
> [Figma link](https://www.figma.com/design/0YfkSvCVtc3Zhh7dBjoMuB/My-Account?node-id=178-5170)
> 
> The screen on the left shows the overview tab with the new QF card. On the right side, you can see the QF card in all different states.
> 
> <img alt="Screenshot 2024-06-24 at 6 16 49 PM" width="537" src="https://private-user-images.githubusercontent.com/13708322/342384776-d54634d4-f6aa-4150-b213-6695d6e85dc3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExOTk5MDMsIm5iZiI6MTcyMTE5OTYwMywicGF0aCI6Ii8xMzcwODMyMi8zNDIzODQ3NzYtZDU0NjM0ZDQtZjZhYS00MTUwLWIyMTMtNjY5NWQ2ZTg1ZGMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDA3MDAwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkOWZkNTEyMzE0NDZlMDc5NTQ1NTNlNjkzMTM3NjgxOWFlZTVkYjY4ODY3ZDIxNmE2ZDZhZGZlNWFmZTc0ODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LXWhXRgGROnTcv9wU5jzhNmkvlx2AaYX36hf3OFO6w4">

The flow that I've seen on staging isn't exactly based on this figme and based on the design in above comment if MDB  failed, the button for checking passport score "connect gitcoin passport" will be enabled,  isn't this design final?


In my test like as below screen by tapping on "check eligibility",  my passport score is shown, how can I be sure that the MDB processing is done ?

https://github.com/user-attachments/assets/19444cb4-bd3a-4077-80d6-f9b6c0dd9016


@Meriem-BM

**Comment by maryjaf** [comment]
- [ ] 2- by tapping on "refresh score"  in QF Donor Eligibility modal , it works only for one time , If the user taps a second time, it will not work

https://github.com/user-attachments/assets/3ef509c4-05ea-4b3a-adba-4b7c0e17bbff

**Comment by Meriem-BM** [comment]
> > Here is the final designs for the **My account / Overview tab**
> > **My account / Overview**
> > [Figma link](https://www.figma.com/design/0YfkSvCVtc3Zhh7dBjoMuB/My-Account?node-id=178-5170)
> > The screen on the left shows the overview tab with the new QF card. On the right side, you can see the QF card in all different states.
> > <img alt="Screenshot 2024-06-24 at 6 16 49 PM" width="537" src="https://private-user-images.githubusercontent.com/13708322/342384776-d54634d4-f6aa-4150-b213-6695d6e85dc3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExOTk5MDMsIm5iZiI6MTcyMTE5OTYwMywicGF0aCI6Ii8xMzcwODMyMi8zNDIzODQ3NzYtZDU0NjM0ZDQtZjZhYS00MTUwLWIyMTMtNjY5NWQ2ZTg1ZGMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDA3MDAwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkOWZkNTEyMzE0NDZlMDc5NTQ1NTNlNjkzMTM3NjgxOWFlZTVkYjY4ODY3ZDIxNmE2ZDZhZGZlNWFmZTc0ODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LXWhXRgGROnTcv9wU5jzhNmkvlx2AaYX36hf3OFO6w4">
> 
> * [ ]  1- The flow that I've seen on staging isn't exactly based on this figme and based on the design in above comment if MDB  failed, the button for checking passport score "connect gitcoin passport" will be enabled,  isn't this design final?
> 
> In my test like as below screen by tapping on "check eligibility", my passport score is shown, how can I be sure that the MDB processing is done ?
> 
>  Screen.Recording.2024-07-17.at.10.23.01.AM.mov 
> @Meriem-BM

Yes, the design is final, is that a new user? that happens because the user has already connected to Gitcoin, before implementing this.

**Comment by maryjaf** [comment]
- [ ] 3- the success donation page isn't based on the design 

![image](https://github.com/user-attachments/assets/5320c0ca-1c5d-493e-ab7e-ba0dbe6b20f9)

Design:

![image](https://github.com/user-attachments/assets/482ce63c-2de7-48c5-b687-12d2617b2bab)

**Comment by Meriem-BM** [comment]
> * [ ]  2- by tapping on "refresh score"  in QF Donor Eligibility modal , it works only for one time , If the user taps a second time, it will not work
> 
>  Screen.Recording.2024-07-17.at.10.56.31.AM.mov

Yes, that happens because of the Cache that Ramin added, pls test by changing your score on Gitcoin and refreshing 2nd time, if it refreshes.

If not I can remove the cache.

**Comment by maryjaf** [comment]
> Yes, the design is final, is that a new user? that happens because the user has already connected to Gitcoin, before implementing this.

I've checked again with a new user 

https://github.com/user-attachments/assets/da03b11b-e024-432e-b015-a861486c95fa

**Comment by maryjaf** [comment]
> Here is the final designs for the **My account / Overview tab**
> **My account / Overview**
> [Figma link](https://www.figma.com/design/0YfkSvCVtc3Zhh7dBjoMuB/My-Account?node-id=178-5170)

@Meriem-BM

**Comment by maryjaf** [comment]
> > * [ ]   2- by tapping on "refresh score"  in QF Donor Eligibility modal , it works only for one time , If the user taps a second time, it will not work
> > 
> > Screen.Recording.2024-07-17.at.10.56.31.AM.mov
> 
> Yes, that happens because of the Cache that Ramin added, pls test by changing your score on Gitcoin and refreshing 2nd time, if it refreshes.
> 
> If not I can remove the cache.

It seems it doesn't work as expected

https://github.com/user-attachments/assets/90683210-ef95-4541-b63b-54c0c06d6d04

**Comment by maryjaf** [comment]
> * [ ]  1- The flow that I've seen on staging isn't exactly based on this figme and based on the design in above comment if MDB  failed, the button for checking passport score "connect gitcoin passport" will be enabled,  isn't this design final?
> 
> In my test like as below screen by tapping on "check eligibility", my passport score is shown, how can I be sure that the MDB processing is done ?


- [ ] 4- It seem this flow in passport banner isn't also based on the[ design ](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail-[-Public-]-(-%E2%9C%85-)?node-id=7012-78&t=vx45Dt02v7nudccH-0)

- I  check with new user 
- I tap on  "check eligibility" button 
- if  MBD  failed , the below banner should be shown based on figma but what i see is like as below screen record 
![image](https://github.com/user-attachments/assets/8502f6e5-a08f-4400-8224-01c9ee5da699)


https://github.com/user-attachments/assets/dc119ca8-7678-4a37-bfda-e21477b2f702

**Comment by Meriem-BM** [comment]
> > * [ ]   1- The flow that I've seen on staging isn't exactly based on this figme and based on the design in above comment if MDB  failed, the button for checking passport score "connect gitcoin passport" will be enabled,  isn't this design final?
> > 
> > In my test like as below screen by tapping on "check eligibility", my passport score is shown, how can I be sure that the MDB processing is done ?
> 
> * [ ]  4- It seem this flow in passport banner isn't also based on the[ design ](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail-%5B-Public-%5D-(-%E2%9C%85-)?node-id=7012-78&t=vx45Dt02v7nudccH-0)
> * I  check with new user
> * I tap on  "check eligibility" button
> * if  MBD  failed , the below banner should be shown based on figma but what i see is like as below screen record
>   ![image](https://private-user-images.githubusercontent.com/111529185/349424722-8502f6e5-a08f-4400-8224-01c9ee5da699.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyMTU4NjcsIm5iZiI6MTcyMTIxNTU2NywicGF0aCI6Ii8xMTE1MjkxODUvMzQ5NDI0NzIyLTg1MDJmNmU1LWEwOGYtNDQwMC04MjI0LTAxYzllZTVkYTY5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxN1QxMTI2MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yY2NhOTBkNTQzODkzMzZiNDVlODVmODNjOTcwNDkwMmQ2YjYxNmNlZDAyYjNhNDFkYjFiNDVkZTcxZWY2ZGE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fHcMLY7mxElhchDw7K_zFs3tCvCMI8pP-O4foIkUL_E)
> 
>  Screen.Recording.2024-07-17.at.11.54.55.AM.mov
 
Yes, it's actually correct but since this user is already connected to Gitcoin, it skipped the Gitcoin connection part to Re-check eligibility where user has to increase his passport score

**Comment by maryjaf** [comment]
> > > * [ ]    1- The flow that I've seen on staging isn't exactly based on this figme and based on the design in above comment if MDB  failed, the button for checking passport score "connect gitcoin passport" will be enabled,  isn't this design final?
> > > 
> > > In my test like as below screen by tapping on "check eligibility", my passport score is shown, how can I be sure that the MDB processing is done ?
> > 
> > 
> > 
> > * [ ]   4- It seem this flow in passport banner isn't also based on the[ design ](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail-%5B-Public-%5D-(-%E2%9C%85-)?node-id=7012-78&t=vx45Dt02v7nudccH-0)
> > * I  check with new user
> > * I tap on  "check eligibility" button
> > * if  MBD  failed , the below banner should be shown based on figma but what i see is like as below screen record
> >   ![image](https://private-user-images.githubusercontent.com/111529185/349424722-8502f6e5-a08f-4400-8224-01c9ee5da699.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyMTU4NjcsIm5iZiI6MTcyMTIxNTU2NywicGF0aCI6Ii8xMTE1MjkxODUvMzQ5NDI0NzIyLTg1MDJmNmU1LWEwOGYtNDQwMC04MjI0LTAxYzllZTVkYTY5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxN1QxMTI2MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yY2NhOTBkNTQzODkzMzZiNDVlODVmODNjOTcwNDkwMmQ2YjYxNmNlZDAyYjNhNDFkYjFiNDVkZTcxZWY2ZGE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fHcMLY7mxElhchDw7K_zFs3tCvCMI8pP-O4foIkUL_E)
> > 
> > Screen.Recording.2024-07-17.at.11.54.55.AM.mov
> 
> Yes, it's actually correct but since this user is already connected to Gitcoin, it skipped the Gitcoin connection part to Re-check eligibility where user has to increase his passport score

I haven't understood this comment

**Comment by Meriem-BM** [comment]
> > * [ ]   1- The flow that I've seen on staging isn't exactly based on this figme and based on the design in above comment if MDB  failed, the button for checking passport score "connect gitcoin passport" will be enabled,  isn't this design final?
> > 
> > In my test like as below screen by tapping on "check eligibility", my passport score is shown, how can I be sure that the MDB processing is done ?
> 
> * [ ]  4- It seem this flow in passport banner isn't also based on the[ design ](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail-%5B-Public-%5D-(-%E2%9C%85-)?node-id=7012-78&t=vx45Dt02v7nudccH-0)
> * I  check with new user
> * I tap on  "check eligibility" button
> * if  MBD  failed , the below banner should be shown based on figma but what i see is like as below screen record
>   ![image](https://private-user-images.githubusercontent.com/111529185/349424722-8502f6e5-a08f-4400-8224-01c9ee5da699.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyMTg0MzUsIm5iZiI6MTcyMTIxODEzNSwicGF0aCI6Ii8xMTE1MjkxODUvMzQ5NDI0NzIyLTg1MDJmNmU1LWEwOGYtNDQwMC04MjI0LTAxYzllZTVkYTY5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxN1QxMjA4NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNTM3NDYwODA0MDRlODUyNTM4ZjFkNDg0NTliNTQyNDg4MTQwYjYwZmQ2ODJkM2MxMTAxM2MyMDNiNjkzYjgyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.f7tqcEHpN6kGAKBUkip4GyuG5HE8zJb6Cu3BACaH8bw)
> 
>  Screen.Recording.2024-07-17.at.11.54.55.AM.mov

This is weird, that user was new user? it actually shouldn't at least show the passport score, it should be 0, will check this.

**Comment by maryjaf** [comment]
> This is weird, that user was new user? it actually shouldn't at least show the passport score, it should be 0, will check this.

Yeah I've tested with new accounts 

- [ ]5-  Now on staging for a user  "0x502677768b396458d63d8c0e146d39d6dd0dcaea"
with passport score:0 and activeQFMBDScore:0, I see 

QF Donor Eligibility : QF Eligible


https://github.com/user-attachments/assets/89ffb524-b86c-4e20-a066-741231f7d434

**Comment by Meriem-BM** [comment]
> > This is weird, that user was new user? it actually shouldn't at least show the passport score, it should be 0, will check this.
> 
> Yeah I've tested with new accounts
> 
> * [ ]   5-  Now on staging for a user  "0x502677768b396458d63d8c0e146d39d6dd0dcaea"
>   with passport score:0 and activeQFMBDScore:0, I see
> 
> QF Donor Eligibility : QF Eligible
> 
>  Screen.Recording.2024-07-17.at.4.34.27.PM.mov

Can Re-check this now I fixed on BE, you can use this updated Link too for testing https://giveth-dapps-v2-9mr4exp2z-givethio.vercel.app/

**Comment by maryjaf** [comment]
> > * [ ]    5-  Now on staging for a user  "0x502677768b396458d63d8c0e146d39d6dd0dcaea"
> [ ] 3- the success donation page isn't based on the design

Thanks @Meriem-BM 
So item 5 and and 3 are resolved? could you please merge it to staging ?

**Comment by maryjaf** [comment]
> [x] 5- Now on staging for a user "0x502677768b396458d63d8c0e146d39d6dd0dcaea"
> [x] 2- by tapping on "refresh score" in QF Donor Eligibility modal , it works only for one time , If the user taps a second time, it will not work

- Item 2 and 5 have been fixed.
- If there is an update on the following items 1,3,4 let me know and I will check again @Meriem-BM 



> * [ ]  1- The flow that I've seen on staging isn't exactly based on this figme and based on the design in above comment if MDB  failed, the button for checking passport score "connect gitcoin passport" will be enabled,  isn't this design final?
> 
> In my test like as below screen by tapping on "check eligibility", my passport score is shown, how can I be sure that the MDB processing is done ?

> [ ] 3- the success donation page isn't based on the design

> * [ ]  4- It seem this flow in passport banner isn't also based on the[ design](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail-%5B-Public-%5D-(-%E2%9C%85-)?node-id=7012-78&t=vx45Dt02v7nudccH-0)

**Comment by laurenluz** [comment]
I checked the user flow for a wallet that passes the MBD... but am a bit confused. What is prompting the dapp to check my address against the API? does it happen just when I sign in? because basically as soon as I signed in & connected, it said I was eligible. @maryjaf @Meriem-BM 

https://github.com/user-attachments/assets/14dfd380-37f9-415b-902b-d3f572637bb7

**Comment by maryjaf** [comment]
> I checked the user flow for a wallet that passes the MBD... but am a bit confused. What is prompting the dapp to check my address against the API? does it happen just when I sign in? because basically as soon as I signed in & connected, it said I was eligible. @maryjaf @Meriem-BM
> 
>  MBD.passed.check.mov

Thanks @laurenluz for checking 
The score of my users are 0 and when @Meriem-BM change my scores manually in admin I see the the same result like you saw and “check eligibility” button was disabled and  I saw my user is qf eligible without tapping in button 
but we thought maybe this behavior is related to manual changes,  so that isn’t true

**Comment by Meriem-BM** [comment]
> I checked the user flow for a wallet that passes the MBD... but am a bit confused. What is prompting the dapp to check my address against the API? does it happen just when I sign in? because basically as soon as I signed in & connected, it said I was eligible. @maryjaf @Meriem-BM
> 
>  MBD.passed.check.mov



> I checked the user flow for a wallet that passes the MBD... but am a bit confused. What is prompting the dapp to check my address against the API? does it happen just when I sign in? because basically as soon as I signed in & connected, it said I was eligible. @maryjaf @Meriem-BM
> 
>  MBD.passed.check.mov

Yes, since we have saved a record of your address for this QF round with a score (=100), once you are logged in, we will pull that record from the backend which is >= the minimum for this QF round, so you are eligible.

If we do not have a record, then the **check eligibility** will appear.

For now we don't have a way to stimulate the process, we change the score manually for the user, maybe we need an address that has (good transaction history) that can have a score > 0

**Comment by Meriem-BM** [comment]
> > Here is the final designs for the **My account / Overview tab**
> > **My account / Overview**
> > [Figma link](https://www.figma.com/design/0YfkSvCVtc3Zhh7dBjoMuB/My-Account?node-id=178-5170)
> > The screen on the left shows the overview tab with the new QF card. On the right side, you can see the QF card in all different states.
> > <img alt="Screenshot 2024-06-24 at 6 16 49 PM" width="537" src="https://private-user-images.githubusercontent.com/13708322/342384776-d54634d4-f6aa-4150-b213-6695d6e85dc3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExOTk5MDMsIm5iZiI6MTcyMTE5OTYwMywicGF0aCI6Ii8xMzcwODMyMi8zNDIzODQ3NzYtZDU0NjM0ZDQtZjZhYS00MTUwLWIyMTMtNjY5NWQ2ZTg1ZGMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDA3MDAwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkOWZkNTEyMzE0NDZlMDc5NTQ1NTNlNjkzMTM3NjgxOWFlZTVkYjY4ODY3ZDIxNmE2ZDZhZGZlNWFmZTc0ODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LXWhXRgGROnTcv9wU5jzhNmkvlx2AaYX36hf3OFO6w4">
> 
> * [x]  1- The flow that I've seen on staging isn't exactly based on this figme and based on the design in above comment if MDB  failed, the button for checking passport score "connect gitcoin passport" will be enabled,  isn't this design final?
> 
> In my test like as below screen by tapping on "check eligibility", my passport score is shown, how can I be sure that the MDB processing is done ?
> 
>  Screen.Recording.2024-07-17.at.10.23.01.AM.mov 
> @Meriem-BM


@laurenluz, @koday1

We have a user who is already connected to Gitcoin (before implementing this feature) but their MBD check failed. Should we still ask them to connect to Gitcoin to follow the same process as the design without skipping that part, or should we directly show his passport score like what happened in the above screen record? WDYT?

cc: @maryjaf

**Comment by Meriem-BM** [comment]
Also @maryjaf I can return an MBD score above threshold "hardcode" to test that scenario for a while then revert it back, WDYT?

**Comment by maryjaf** [comment]
> Also @maryjaf I can return an MBD score above threshold "hardcode" to test that scenario for a while then revert it back, WDYT?

Good suggestion, I agree

**Comment by maryjaf** [comment]
remaining items :

For these two things a decision should be made by product team

cc: @Meriem-BM @jainkrati @laurenluz @koday1 


- [ ] 1-

> @laurenluz, @koday1
> 
> We have a user who is already connected to Gitcoin (before implementing this feature) but their MBD check failed. Should we still ask them to connect to Gitcoin to follow the same process as the design without skipping that part, or should we directly show his passport score like what happened in the above screen record? WDYT?
> 
> cc: @maryjaf

- [ ] 2-

> > I checked the user flow for a wallet that passes the MBD... but am a bit confused. What is prompting the dapp to check my address against the API? does it happen just when I sign in? because basically as soon as I signed in & connected, it said I was eligible. @maryjaf @Meriem-BM
> > MBD.passed.check.mov
> 
> > I checked the user flow for a wallet that passes the MBD... but am a bit confused. What is prompting the dapp to check my address against the API? does it happen just when I sign in? because basically as soon as I signed in & connected, it said I was eligible. @maryjaf @Meriem-BM
> > MBD.passed.check.mov
> 
> Yes, since we have saved a record of your address for this QF round with a score (=100), once you are logged in, we will pull that record from the backend which is >= the minimum for this QF round, so you are eligible.
> 
> If we do not have a record, then the **check eligibility** will appear.
> 
> For now we don't have a way to stimulate the process, we change the score manually for the user, maybe we need an address that has (good transaction history) that can have a score > 0

**Comment by laurenluz** [comment]
> We have a user who is already connected to Gitcoin (before implementing this feature) but their MBD check failed. Should we still ask them to connect to Gitcoin to follow the same process as the design without skipping that part, or should we directly show his passport score like what happened in the above screen record? WDYT?

@Meriem-BM @maryjaf 

We need to make sure that what our dapp has recorded as the person's gitcoin passport score is recent (i.e. it was checked during the active QF round). So:
- if the user had connected their gitcoin passport before we implemented this, AND
- the last time we "checked" their passport score was sometime recently (i.e. since the start of the current QF round), AND
- the fail the MBD check

Then we can skip the "connect passport" step, and just show them whatever prompt based on their score... i.e. "you're eligible!" if their score is above the passport threshold for the round, and "increase your score" if their score is below the passport threshold for the round.

> Yes, since we have saved a record of your address for this QF round with a score (=100), once you are logged in, we will pull that record from the backend which is >= the minimum for this QF round, so you are eligible.
> 
> If we do not have a record, then the check eligibility will appear.
> 
> For now we don't have a way to stimulate the process, we change the score manually for the user, maybe we need an address that has (good transaction history) that can have a score > 0

I'm ok with this flow! If you check my address's score and compare against our threshold for the MBD system upon sign-in... and my score is pass, it's ok to just tell the user right away that they are ok.

I can perform the tests we need using my address to check that all the MBD stuff works ok. And also @koday1 I bet can also be a 2nd tested, as I think his main address probably has a passing score too.

btw - what is the threshold score that a user must pass to be considered a "pass" against the MBD system? @Meriem-BM @CarlosQ96

**Comment by Meriem-BM** [comment]
> btw - what is the threshold score that a user must pass to be considered a "pass" against the MBD system? @Meriem-BM @CarlosQ96

There is a table in the docs that explains it (Section: ETH Mainnet Activity Model) page 4 

[[Beta] Model Based Detection API.pdf](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNjRmQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--8df727b78b70157f0f0a564c9e4c04df6f45e1ae/%5BBeta%5D%20Model%20Based%20Detection%20API.pdf)

Basically it's between 0 and 100.

**Comment by laurenluz** [comment]
Yeah :) But we should be able to set our threshold for "passing score" via the MBD system in the backend. Actually, we should be able to set the passing score threshold in admin bro corresponding to each round. - I'm not sure if this was implemented.

So for example, we set the MBD threshold to be 15... and then users with 14 need to fill out passport... and users with score about 15 pass the MBD. @Meriem-BM

**Comment by Meriem-BM** [comment]
> Yeah :) But we should be able to set our threshold for "passing score" via the MBD system in the backend. Actually, we should be able to set the passing score threshold in admin bro corresponding to each round. - I'm not sure if this was implemented.
> 
> So for example, we set the MBD threshold to be 15... and then users with 14 need to fill out passport... and users with score about 15 pass the MBD. @Meriem-BM

Oh, my bad, yes it's there, for this active round it's `0.9` you can find it here https://impact-graph.serve.giveth.io/admin/resources/QfRound/records/90/show as `Minimum User Analysis Score`

**Comment by laurenluz** [comment]
Btw, we shouldn't have a "qf-eligibility" page, i.e. https://staging.giveth.io/qf-eligibility

When the user clicks "check eligibility" and need to go through the designed flow, it should all be in modals. Right now on staging, after donating, I clicked "check eligibility" and ended up on this page that shouldn't exists 😅 

We should only show these parts as modals or in the user's "my account"

![image](https://github.com/user-attachments/assets/3081a709-c56f-42e3-93ce-f3da482290ce)

![image](https://github.com/user-attachments/assets/1086ca22-2e12-4f37-b9d4-308db042bd78)

**Comment by laurenluz** [comment]
> Oh, my bad, yes it's there, for this active round it's 0.9 you can find it here https://impact-graph.serve.giveth.io/admin/resources/QfRound/records/90/show as Minimum User Analysis Score

Can we change the name to ```Min MBD Score```? I think it will be easier for onboarding admins in the future

![image](https://github.com/user-attachments/assets/cc73183b-8e9c-4461-96de-35c72f1c6f93)

**Comment by Meriem-BM** [comment]
> We should only show these parts as modals or in the user's "my account"

Oh, I can change that, should I redirect him to account page, it would be easier

**Comment by Meriem-BM** [comment]
> FYI I tested the one-click "check eligibility" from the qf eligibily page and it worked as expected with my MBD score >0 wallet.
> 
> Should I re-test the flow from the banner? @Meriem-BM maybe you can delete my score again in the backend?

Yes, deleted, you can test the other flow (failed MBD) by modifying the threshold to >= 71 (yours is 70 now)

**Comment by Meriem-BM** [comment]
> Can we change the name to `Min MBD Score`? I think it will be easier for onboarding admins in the future

@CarlosQ96

**Comment by CarlosQ96** [comment]
Yeah we can change the name @Meriem-BM

Checking all @laurenluz  feedback

**Comment by laurenluz** [comment]
> Oh, I can change that, should I redirect him to account page, it would be easier

@mosaeedi put in the design flow to just open modals when the user has to go through it... so they don't get taken away from the donating page or the projects page... and they don't lose track of what they are doing.

The user should be able to do the check flow through my account.. but we shouldn't send them there from the banners or the donation success page.

**Comment by Meriem-BM** [comment]
> @mosaeedi put in the design flow to just open modals when the user has to go through it... so they don't get taken away from the donating page or the projects page... and they don't lose track of what they are doing.
> 
> The user should be able to do the check flow through my account.. but we shouldn't send them there from the banners or the donation success page.

Seems like I didn't notice this success page flow while working :), I'll update it soon, by adding the model.

**Comment by laurenluz** [comment]
Thank you @Meriem-BM !

**Comment by Meriem-BM** [comment]
> @mosaeedi put in the design flow to just open modals when the user has to go through it... so they don't get taken away from the donating page or the projects page... and they don't lose track of what they are doing.
> 
> The user should be able to do the check flow through my account.. but we shouldn't send them there from the banners or the donation success page.
 
@laurenluz the modal is added now as well as the filed is renamed to `Min MBD Score`

**Comment by maryjaf** [comment]
Thanks @Meriem-BM 

- The change of name is passed
- Showing modal by tapping on "check eligibility" button in success donation page, is passed 


https://github.com/user-attachments/assets/31bce25b-2439-45a5-a71f-dfc4ca435507

**Comment by maryjaf** [comment]
When there is no active round, the "QF eligibility check" section isn't shown in my account.
I'm not sure is this an expected result or not, because in previous model user could be able to check his score when there was no active round 

![image](https://github.com/user-attachments/assets/c136110f-a3df-4e4d-9ce7-068170f2fee2)

@Meriem-BM

**Comment by maryjaf** [comment]
So I think it would be good some changes should be made in this page when there is no active round  https://giveth.io/qf-eligibility

cc: @Meriem-BM @mosaeedi 

![image](https://github.com/user-attachments/assets/32390475-e646-41f4-9d97-7e3103420bdb)

**Comment by laurenluz** [comment]
@maryjaf @Meriem-BM - there should be no page " https://giveth.io/qf-eligibility"

the final design has this flow all in modals, or in "my acconut", but we should just remove the qf-eligibility page (and /passport page) entirely.

cc @mosaeedi

**Comment by maryjaf** [comment]
Thanks @Meriem-BM 
- This page has been removed " https://giveth.io/qf-eligibility"

**Comment by maryjaf** [comment]
I've checked different scenarios on STG and they work as expected 

- User has MBD score and passport score lower than threshold 
My account view: pass
Passport banner: pass
Success donation page : pass

- User has MBD score  lower than threshold  and passport score higher than threshold 
My account view: pass
Passport banner: pass
Success donation page : pass

- User has MBD score  higher  than threshold  
My account view: pass
Passport banner: pass
Success donation page : pass

**Comment by laurenluz** [comment]
I asked @WhyldWanderer to do a test with this, with a wallet that had MBD score lower than the threshold... and she had an issue...

Basically there was some weirdness where when she clicks "check eligibility" after connecting her passport, nothing happens...

Here is a screen record @Meriem-BM - can you check?
https://github.com/user-attachments/assets/e6ea9881-ec56-4ba3-8ef0-7c5124c1d617

**Comment by mosaeedi** [comment]
## Low Passport score modal

When user connect their Gitcoin passport and they have a low score, we need to change the copy to tell them what they need to do, and also add a button to open Gitcoin passport `https://passport.gitcoin.co/` on a new tab.

Please take a look at the modal on Figma -> [Figma Link](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail-%5B-Public-%5D-(-%E2%9C%85-)?node-id=7035-4191&t=cZYTYIgHK3RjQ1E2-4)

<img width="644" alt="Screenshot 2024-07-24 at 4 23 51 PM" src="https://github.com/user-attachments/assets/bdc9d7bd-c2b9-4289-8ebb-868acff5780a">

**Comment by maryjaf** [comment]
- [ ] The modal is closed automatically after multis secs I think it should be closed by user tapping 

https://github.com/user-attachments/assets/f80bf43b-9ec3-480a-8c84-177081611058

**Comment by laurenluz** [comment]
Almond tried testing this also FYI about 14 hrs ago and said:
"I tried it. I think it works. It’s kind of buggy tho. I got a message I was not eligible - then I clicked on the verify your stamps button, but nothing happened. Then I refreshed and it said, I was eligible." 🤔

**Comment by maryjaf** [comment]
> Almond tried testing this also FYI about 14 hrs ago and said: "I tried it. I think it works. It’s kind of buggy tho. I got a message I was not eligible - then I clicked on the verify your stamps button, but nothing happened. Then I refreshed and it said, I was eligible." 🤔

It works like as below screen record for me , Do you have a record of this error ?

https://github.com/user-attachments/assets/b5b975ba-25f0-4bbe-905f-bb6f7e561409

**Comment by Meriem-BM** [comment]
Sorry for not merging changes yesterday, It's on staging now, you can test it @maryjaf 

cc: @laurenluz

**Comment by maryjaf** [comment]
> ## Low Passport score modal
> When user connect their Gitcoin passport and they have a low score, we need to change the copy to tell them what they need to do, and also add a button to open Gitcoin passport `https://passport.gitcoin.co/` on a new tab.
> 
> Please take a look at the modal on Figma -> [Figma Link](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail-%5B-Public-%5D-(-%E2%9C%85-)?node-id=7035-4191&t=cZYTYIgHK3RjQ1E2-4)
> 
> <img alt="Screenshot 2024-07-24 at 4 23 51 PM" width="644" src="https://private-user-images.githubusercontent.com/13708322/351721434-bdc9d7bd-c2b9-4289-8ebb-868acff5780a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5ODkxOTQsIm5iZiI6MTcyMTk4ODg5NCwicGF0aCI6Ii8xMzcwODMyMi8zNTE3MjE0MzQtYmRjOWQ3YmQtYzJiOS00Mjg5LThlYmItODY4YWNmZjU3ODBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI2VDEwMTQ1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhODNjZjgxZDJhOTAwNTkxOTIxODQwNjgxMzBkMDFmMGZhN2E1YmRmODdlNzFjYWFmNmNiYTRmZDQwYzZhODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4mhYqD0lXpD-8jMs5rnfR5JJlhd_nV0K_YUfy9omj60">

When the user score is more than threshold , "increase your score" button should be shown?

![image](https://github.com/user-attachments/assets/be32b784-e37b-4409-8265-29d9acd5025c)
cc: @laurenluz @mosaeedi

**Comment by laurenluz** [comment]
Some small bugs/weirdnesses I found:

- [ ] When I connected a new wallet (that I never used before to check my score)... it right away sent me to "We need a bit more info" (i.e. failed, MBD, passport not connected). Is that expected? I thought we didn't have any "auto-check MBD on sign in"? (you can see this in the video below)

- [ ] Upon connecting my gitcoin passport the 1st time, when it loaded my passport the modal just auto-closed without me doing anything

https://github.com/user-attachments/assets/b8591959-73bd-48e4-99a2-fbfbcaf8c0a2

**Comment by maryjaf** [comment]
> * [x]  The modal is closed automatically after multi secs I think it should be closed by user tapping


> * Upon connecting my gitcoin passport the 1st time, when it loaded my passport the modal just auto-closed without me doing anything

It seems this problem has been fixed in new changes, I faced this problem recently  but in my new test it has been fixed

**Comment by laurenluz** [comment]
There is definitely something funny happening when the user connects a wallet newly for the first time. I just watched Franco test it here (my roommate :-D)... he was doing the test with the spanish version of the dapp, but the flow was also interesting...

1. connected his wallet
2. saw the banner "we need a bit more information to validate your qf eligibility" ( i.e. as if he clicked & failed the MBD.. but he hadn't).
3. clicked to add more info, connected his passport and signed the message...
4. the modal auto-closed (weird and buggy)
5. he got the banner "verify you eligibility with one click" (i.e. as if he was being now checked against the MBD system)...
6. he clicked the "one click check"... and he got the "pass"

I don't know if his pass was because of his gitcoin passport, or because of the MBD system, but this wasn't the flow I expected.... even though it actually kind of made sense 😓

**Comment by Meriem-BM** [comment]
> > ## Low Passport score modal
> > When user connect their Gitcoin passport and they have a low score, we need to change the copy to tell them what they need to do, and also add a button to open Gitcoin passport `https://passport.gitcoin.co/` on a new tab.
> > Please take a look at the modal on Figma -> [Figma Link](https://www.figma.com/design/Qp56yzWMCb88SofnuhXhu3/Project-Detail-%5B-Public-%5D-(-%E2%9C%85-)?node-id=7035-4191&t=cZYTYIgHK3RjQ1E2-4)
> > <img alt="Screenshot 2024-07-24 at 4 23 51 PM" width="644" src="https://private-user-images.githubusercontent.com/13708322/351721434-bdc9d7bd-c2b9-4289-8ebb-868acff5780a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5ODkxOTQsIm5iZiI6MTcyMTk4ODg5NCwicGF0aCI6Ii8xMzcwODMyMi8zNTE3MjE0MzQtYmRjOWQ3YmQtYzJiOS00Mjg5LThlYmItODY4YWNmZjU3ODBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI2VDEwMTQ1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhODNjZjgxZDJhOTAwNTkxOTIxODQwNjgxMzBkMDFmMGZhN2E1YmRmODdlNzFjYWFmNmNiYTRmZDQwYzZhODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4mhYqD0lXpD-8jMs5rnfR5JJlhd_nV0K_YUfy9omj60">
> 
> When the user score is more than threshold , "increase your score" button should be shown?
> 
> ![image](https://private-user-images.githubusercontent.com/111529185/352496171-be32b784-e37b-4409-8265-29d9acd5025c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5OTEyOTEsIm5iZiI6MTcyMTk5MDk5MSwicGF0aCI6Ii8xMTE1MjkxODUvMzUyNDk2MTcxLWJlMzJiNzg0LWUzN2ItNDQwOS04MjY1LTI5ZDlhY2Q1MDI1Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyNlQxMDQ5NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZTc1ODYxMzRlOTFhZTJlZWU2ZWM1N2E1OTUwOTQxMmFkNzk4NGFiZDg1NGU4OGRkNGI4OTlkOGZlY2Q1NzBmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QI5Ke-dTYCzX-NN5jvGLVxCmy2mkNNLiMwcL5fKdx4k) cc: @laurenluz @mosaeedi @Meriem-BM

Thanks, I'll change it

**Comment by Meriem-BM** [comment]
> There is definitely something funny happening when the user connects a wallet newly for the first time. I just watched Franco test it here (my roommate :-D)... he was doing the test with the spanish version of the dapp, but the flow was also interesting...
> 
> 1. connected his wallet
> 2. saw the banner "we need a bit more information to validate your qf eligibility" ( i.e. as if he clicked & failed the MBD.. but he hadn't).
> 3. clicked to add more info, connected his passport and signed the message...
> 4. the modal auto-closed (weird and buggy)
> 5. he got the banner "verify you eligibility with one click" (i.e. as if he was being now checked against the MBD system)...
> 6. he clicked the "one click check"... and he got the "pass"
> 
> I don't know if his pass was because of his gitcoin passport, or because of the MBD system, but this wasn't the flow I expected.... even though it actually kind of made sense 😓

I think this because of cache, which was changed recently, I also got same issue and even get signed out

**Comment by laurenluz** [comment]
> When the user score is more than threshold , "increase your score" button should be shown?

I would say no!! No need to send them to increase their score if their score is already enough @Meriem-BM @maryjaf

**Comment by laurenluz** [comment]
> I think this because of cache, which was changed recently, I also got same issue and even got signed out

So we can generally expect this won't happen on production? @Meriem-BM

**Comment by laurenluz** [comment]
Ashley tested and has some strange flow as well (similar to Franco) and took a screenrecord.

The behaviour:
- user with a MBD too low, opens the modal and connects their passport.
- After signing the passport message, the modal closes automatically (this shouldn't happen, the user should instead be able to see what their passport score and the right message depending on if it is above or below the threshold, and the user should close the modal or click away themself)
- the banner says "verify your uniqueness with one click" (this is the wrong banner for this stage. the user who has already signed the message to connect their passport here should see the banner that either says "increase your score" or "you are eligible" after the connected their passport)

See below buggy flow:
https://github.com/user-attachments/assets/794a612f-6317-456d-8fcb-7f5bcda2b511

**Comment by laurenluz** [comment]
Hey @Meriem-BM a couple of points we talked about fixing in DM:

- [ ] To remedy the issues on the connect wallet / sign in from the banner... let's just remove the "connect wallet" button from the (not signed in/connected) banner and say "Sign in with your wallet to verify your eligibility"

- [ ] Let's make sure that we hide the qf-eligibility section in "My account" if the round is before the "start date" and after the "end date". We should only have the qf-eligibility section during an active qf round that is past the start date  and before the end date

**Comment by maryjaf** [comment]
The flow related to checking score for a new user with mbd and passport score lower than threshold
and it seems the flow is based on figma

https://github.com/user-attachments/assets/87c1e545-c76b-428e-9a3a-73f623f42ace

Should sth be changed in this flow or not ? I am asking because I have not been online for the past few hours and I may have missed an update

**Comment by laurenluz** [comment]
@maryjaf that flow is perfect!

**Comment by maryjaf** [comment]
> * [ ]  Let's make sure that we hide the qf-eligibility section in "My account" if the round is before the "start date" and after the "end date". We should only have the qf-eligibility section during an active qf round that is past the start date  and before the end date

It has been fixed on feature branch https://giveth-dapps-v2-git-feat-mbdscoringissues-givethio.vercel.app


> * To remedy the issues on the connect wallet / sign in from the banner... let's just remove the "connect wallet" button from the (not signed in/connected) banner and say "Sign in with your wallet to verify your eligibility"

The new flow with "sign message" step is on  the preview link : https://giveth-dapps-v2-git-feat-mbdscoringissues-givethio.vercel.app

https://github.com/user-attachments/assets/a8e50aed-5651-446f-9bd0-37ddf9fb0b20


There is a problem about this flow that I couldn't found it the reason of this 

- For an account that it isn't signed user, after connecting  wallet the "sign message" is shown correctly 

https://github.com/user-attachments/assets/d7d11a21-67b8-4aa8-9db3-96db38775dc1


-For this account that it isn't signed user, after connecting  wallet the "sign message" is not  shown 
https://github.com/user-attachments/assets/8f1e2b31-43f0-4322-9a86-402502cf4b5d

**Comment by maryjaf** [comment]
- [ ] On production there is a problem that for all users the below pic is shown in my account and the "check eligibility" button isn't shown 

when changes will be deployed on production?

![image](https://github.com/user-attachments/assets/69b84580-e5d0-44fe-ba09-0ba3a533a5d1)

**Comment by Meriem-BM** [comment]
> > * [ ]   Let's make sure that we hide the qf-eligibility section in "My account" if the round is before the "start date" and after the "end date". We should only have the qf-eligibility section during an active qf round that is past the start date  and before the end date
> 
> It has been fixed on feature branch https://giveth-dapps-v2-git-feat-mbdscoringissues-givethio.vercel.app
> 
> > * To remedy the issues on the connect wallet / sign in from the banner... let's just remove the "connect wallet" button from the (not signed in/connected) banner and say "Sign in with your wallet to verify your eligibility"
> 
> The new flow with "sign message" step is on the preview link : https://giveth-dapps-v2-git-feat-mbdscoringissues-givethio.vercel.app
> 
>  Screen.Recording.2024-07-27.at.12.50.30.AM.mov 
> * [ ]  There is a problem about this flow that I couldn't found it the reason of this
> 
> -For an account that it isn't signed user, after connecting wallet the "sign message" is shown correctly
> 
>  Screen_Recording_2024-07-27_at_12.59.06_AM.mov 
> -For this account that it isn't signed user, after connecting wallet the "sign message" is not shown https://github.com/user-attachments/assets/8f1e2b31-43f0-4322-9a86-402502cf4b5d

Hey @maryjaf this only happens when a new user is connecting, the sign message add his data to Database, without the user data won't be found and that buggy behaviour appears, I believe we could have implemented this in better way to have one step, but that will introduce bugs.

**Comment by Meriem-BM** [comment]
> * [ ]  On production there is a problem that for all users the below pic is shown in my account and the "check eligibility" button isn't shown
> 
> when changes will be deployed on production?
> 
> ![image](https://private-user-images.githubusercontent.com/111529185/352798098-69b84580-e5d0-44fe-ba09-0ba3a533a5d1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxNjAwNzAsIm5iZiI6MTcyMjE1OTc3MCwicGF0aCI6Ii8xMTE1MjkxODUvMzUyNzk4MDk4LTY5Yjg0NTgwLWU1ZDAtNDRmZS1iYTA5LTBiYTNhNTMzYTVkMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyOFQwOTQyNTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYmZhZDU0ZDY3ODJhMTY0YzNjODhlNWE2YjZlZDM5MmU5NTRkYjlmMjg2ODhmNjNhNjE4ZjE1ZGE0ODg5NTk4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3JbsPXMT0Kq0xQfKe9OxsHw9kF81hOmv4lpeEvCoM78)

I think we should discuss this with the team, if we should only release QF related changes.

cc: @jainkrati, @MohammadPCh

**Comment by laurenluz** [comment]
> Hey @maryjaf this only happens when a new user is connecting, the sign message add his data to Database, without it, the user data won't be found and that buggy behaviour appears, I believe we could have implemented this in better way to have one step, but that will introduce bugs.

Can we just remove any mention of connect wallet & sign message for the user? so they just follow the normal slow having to sign in like usual? I mentioned this here above, but I think it might have been misinterpreted...

>  To remedy the issues on the connect wallet / sign in from the banner... let's just remove the "connect wallet" button from the (not signed in/connected) banner and say "Sign in with your wallet to verify your eligibility"

I meant that we should *just* put the copy "Sign in with your wallet to verify your QF eligibility".. with absolutely no button there/clickable part in the banner... so no extra clickable text that says "sign in" or "sign message" or "connect wallet"

The user should only get to click things after they have done their normal sign in flow. with the pink button in the top (like usual).

If there are still bugs, can we just simplify to this?

**Comment by Meriem-BM** [comment]
@laurenluz that won't work either, because by clicking on that pink sign in button the new user will still have to sign message, to be registered on DB.

The pink sign on button only connects wallet as well.

**Comment by laurenluz** [comment]
@Meriem-BM why is the sign message about gitcoin passport though?

I would assume it should be about connecting to giveth

**Comment by laurenluz** [comment]
this is still super buggy for me. When I connect my wallet from the banner... I get the metamask connect... but then nothign happens, even when I refresh. 

This issue doesn't happen if I connect & sign in from the pink button.

https://github.com/user-attachments/assets/8464db19-00c3-4d51-92c3-5f8fb83220f3

**Comment by maryjaf** [comment]
> this is still super buggy for me. When I connect my wallet from the banner... I get the metamask connect... but then nothign happens, even when I refresh.
> 
> This issue doesn't happen if I connect & sign in from the pink button.
> 
>  Screen.Recording.2024-07-29.at.1.34.07.PM.mov

It's weird, it works fine for me,I've checked multiple times with different accounts
 Do you know what is the reason of this behavior ? @Meriem-BM

**Comment by Meriem-BM** [comment]
> this is still super buggy for me. When I connect my wallet from the banner... I get the metamask connect... but then nothign happens, even when I refresh.
> 
> This issue doesn't happen if I connect & sign in from the pink button.
> 
>  Screen.Recording.2024-07-29.at.1.34.07.PM.mov

Weird, I don't know why did it happen

**Comment by Meriem-BM** [comment]
Hey @maryjaf the connect wallet clickable link is removed.
Also can you test the the banner when round is not started yet, if we should change something, Thanks.

cc: @laurenluz

**Comment by maryjaf** [comment]
> Also can you test the the banner when round is not started yet, if we should change something, Thanks.

- [ ] It seems it hasn't been changed as expected, the round will start on 1 Aug but I see below banner 

![image](https://github.com/user-attachments/assets/60863b09-2ab7-4a21-ae4d-5d401a86544a)

**Comment by maryjaf** [comment]
> * [ ]  It seems it hasn't been changed as expected, the round will start on 1 Aug but I see below banner

and also when there is no active round or the round has ended, this banner shouldn't been shown

![image](https://github.com/user-attachments/assets/e4cbd81a-237a-4730-82d1-0026967ace87)

**Comment by Meriem-BM** [comment]
Okay so all above should be done before user connect his wallet, unlike the old flow, will update it.

**Comment by maryjaf** [comment]
It seems this problem makes some difficulty for QF round
https://github.com/Giveth/giveth-dapps-v2/issues/4377 

https://github.com/user-attachments/assets/527b9879-9cf5-4ca5-ba3b-9cdcb80d6cbf

 @divine-comedian @laurenluz

**Comment by laurenluz** [comment]
@Meriem-BM can you investigate? that's super weird behaviour

**Comment by maryjaf** [comment]
> Hey @maryjaf the connect wallet clickable link is removed.

 I've checked this new flow with some new and old users for most of them it works as expected

https://github.com/user-attachments/assets/0a45cd4a-2a5c-4581-a7bc-aa5f4cf6d655


- [ ] but I see this the below behavior for a new user after sign

-by tapping on "check eligibility" nothing happens and then the "recheck eligibility "  button is shown

https://github.com/user-attachments/assets/7c9bd9ef-2198-4d0f-b852-29f3a3551b57

@Meriem-BM

**Comment by laurenluz** [comment]
@Meriem-BM just checking - any update here?

**Comment by maryjaf** [comment]
> * [ ]  but I see this the below behavior for a new user after sign
> 
> -by tapping on "check eligibility" nothing happens and then the "recheck eligibility " button is shown

I've seen this problem one time in my tests, could you please take a final look on this ? @Meriem-BM 
If needed I can create separate issue for this to be investigated more

**Comment by Meriem-BM** [comment]
Thanks @maryjaf, can you create a separate issue for this with details, will try to check or ask someone if possible.
